app/src/main/java/cm/aptoide/pt/app/view/AppViewFragment.java

```diff
@@ -1673,14 +1678,21 @@ private SpannableString formatAppCoinsRewardMessage(String rewardValue) {
     return appBought;
   }
 
-  private void showAppcInfo(boolean hasAdvertising, boolean hasBilling, String appcReward,
-      double fiatReward, String fiatCurrency) {
+  private void showAppcInfo(boolean hasAdvertising, boolean hasBilling, double appcReward,
+      double fiatReward, String fiatCurrency, double appcBudget) {
     if (hasAdvertising) {
       String formatedFiatCurrency = fiatCurrency + poaFiatDecimalFormat.format(fiatReward);
       appcInfoView.setVisibility(View.VISIBLE);
       poaOfferValue.setText(
-          String.format(getResources().getString(R.string.poa_app_view_card_body_2), appcReward,
-              formatedFiatCurrency));
+          String.format(getResources().getString(R.string.poa_app_view_card_body_2),
+              String.valueOf(appcReward), formatedFiatCurrency));
+      if (appcBudget != -1.0) {
```

Comment:
Does the budget only appear if it's a FOMO? I.e., is the FOMO rule in the API side? Looks like it's a yes since it works, but idk.

Reply:
Yeah, the rule is on the API side. Found this out by crashing the app when it didn't come, hence these rules on our side too;